xlators/protocol/client/src/client-rpc-fops.c

```diff
@@ -2401,6 +2402,26 @@ client3_3_lk_cbk(struct rpc_req *req, struct iovec *iov, int count,
         goto out;
     }
 
+    xdata = dict_new();
+    if (xdata == NULL) {
+        rsp.op_errno = ENOMEM;
+        rsp.op_ret = -1;
+        goto out;
+    }
+
+    if (local->cmd == F_GETLK) {
+        conf = this->private;
+        if (!conf)
+            goto out;
+
+        if (conf->strict_locks) {
+            ret = dict_set_int32(xdata, "strict-locks", 1);
+            if (ret)
+                gf_msg_debug(THIS->name, ENOMEM,
+                             "Failed to set strict-locks status into dict");
+        }
+    }
+
     if (rsp.op_ret >= 0) {
         ret = client_post_lk(this, &rsp, &lock, &xdata);
```

Comment:
Same comment as below about xdata getting overwritten. Also this is code duplication. Could you make this into a function?

Reply:
Done.